### ISSUE TITLE
Bug: [@typescript-eslint/no-unsafe-call] Error for jest tests defined with it.each()

### ISSUE THREAD

**Issue by Ptacnik** [typescript-eslint/typescript-eslint#6413]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play#ts=4.9.3&sourceType=module&code=MYewdgzgLgBANgS2gLhtATgsBzA2gXRgF4ZcAiAQTIBoZKAhM-AbgFgAoDhKAOgFMAhsAAWACkTQAlKLLo+UAK7pIMKOgV8YAMxDoYANwGIAJjDACAtnwg0Yo81cnEAfDADeHGDD4APAA58wFCiECBWALLywiDG9pZ8kpI8UCD0fAAq6lDCAJ6ikmzsAL4FHBxaCmBBCOBoYXyR2TFxVqgYWNhOHuxecorKqup8hUVAA&eslintrc=N4KABGBEBOCuA2BTAzpAXGYkACAXAngA4oDG0AlobgLQrzkB2uA9AwPbWwPICGAZomoke8eOiiJo0NtEgBfEHKA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA

### Repro Code

```typescript
const list: string[] = ["A", "AB"];

it.each(list)("returns true for valid names", (name) => {
  expect(someMethod(name)).toBeTruthy();
});

function someMethod(name: string) {
  return true;
}
```

### ESLint Config

```javascript
{
  "rules": {"@typescript-eslint/no-unsafe-call": "error"}
}
```

### Expected Result

I expected that this unit test method will not be a problem for eslint validation.

### Actual Result

Eslint is producing error.

Warning in VSC:
![image](https://user-images.githubusercontent.com/3058114/216375974-3b41dc04-ac0c-4184-80b6-7f31448d6b5a.png)

**Comment by Josh-Cena** [comment]
Have you tried restarting your editor and/or running ESLint CLI and see if the error persists?

**Comment by bradzacher** [comment]
If this happens with the rule, generally it means that the types are missing in some way - I.e. Our tooling can't find the types, so TS defaults to `any`, so the rules flag the `any`.

Without more information it's hard to say what the issue would be within your codebase.

**Comment by c-vetter** [comment]
I had this issue in relation to https://github.com/cypress-io/cypress/issues/1603
in a project trying to use https://github.com/PublicisSapient/cra-template-accelerate

The proposed solution doesn't seem to work in any permutation I've found except when removing cypress.

I'm giving this up for now, but maybe this helps somebody else.

**Comment by Ptacnik** [comment]
> Have you tried restarting your editor and/or running ESLint CLI and see if the error persists?

@Josh-Cena yes. I've tried that before raising this topic.

**Comment by Ptacnik** [comment]
> If this happens with the rule, generally it means that the types are missing in some way - I.e. Our tooling can't find the types, so TS defaults to `any`, so the rules flag the `any`.
> 
> Without more information it's hard to say what the issue would be within your codebase.

I'll try to reproduce on codes that I can share. But I'll follow your answer and dig a little more.

**Comment by amitkumarhq** [comment]
![image](https://user-images.githubusercontent.com/94821119/221680975-e4e88b14-e908-4aea-994a-2a125544c480.png)
I'm also facing a similar issue in a Next.js app, but it is weird because the error only shows when I import the `navLinks` variable from a different file. The error vanishes while performing the function call in the same file where the variable is initialized.
![image](https://user-images.githubusercontent.com/94821119/221682143-4555929f-6dbb-4222-bfac-7001fdfa7684.png)

**Comment by Ptacnik** [comment]
Hello.

Sorry for delays. I had time to check it out and figure out, where the problem was. I had missing import for it - now it works.

`import { expect, it } from "@jest/globals";`

This is not a bug but mistake on my side -> closing this issue.
